daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/engine/EngineTest.scala

```diff
@@ -795,6 +706,67 @@ class EngineTest
           )
       }
     }
+
+    "unused disclosed contracts not saved to ledger" in {
+      val templateId = Identifier(basicTestsPkgId, "BasicTests:WithKey")
+      val usedContractSKey = SValue.SRecord(
```

Comment:
Instead of defining the two separately maybe consider deriving one from the other? (Probably svalue from the value)